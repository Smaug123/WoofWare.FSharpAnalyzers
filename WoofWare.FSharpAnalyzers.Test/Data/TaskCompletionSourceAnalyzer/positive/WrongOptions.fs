module TaskCompletionSourceWrongOptions

open System.Threading.Tasks

let createTcsWithNone () =
    let tcs = TaskCompletionSource<int> TaskCreationOptions.None
    tcs.SetResult 42
    tcs.Task

let createTcsWithDenyChildAttach () =
    let tcs = TaskCompletionSource<string> TaskCreationOptions.DenyChildAttach
    tcs.SetResult "hello"
    tcs.Task

let createTcsWithMultipleWrongOptions () =
    let tcs =
        TaskCompletionSource<int> (TaskCreationOptions.DenyChildAttach ||| TaskCreationOptions.PreferFairness)

    tcs.SetResult 1
    tcs.Task
