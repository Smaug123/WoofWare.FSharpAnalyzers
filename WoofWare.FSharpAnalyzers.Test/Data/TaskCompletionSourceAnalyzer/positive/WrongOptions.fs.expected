WOOF-TCS-ASYNC | Warning | (6,14--6,64) | TaskCompletionSource<T> created without TaskCreationOptions.RunContinuationsAsynchronously. This can cause continuations to run inline on the calling thread, leading to deadlocks, thread-pool starvation, and corruption of state. Always use: new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously).
WOOF-TCS-ASYNC | Warning | (11,14--11,78) | TaskCompletionSource<T> created without TaskCreationOptions.RunContinuationsAsynchronously. This can cause continuations to run inline on the calling thread, leading to deadlocks, thread-pool starvation, and corruption of state. Always use: new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously).
WOOF-TCS-ASYNC | Warning | (17,8--17,110) | TaskCompletionSource<T> created without TaskCreationOptions.RunContinuationsAsynchronously. This can cause continuations to run inline on the calling thread, leading to deadlocks, thread-pool starvation, and corruption of state. Always use: new TaskCompletionSource<T>(TaskCreationOptions.RunContinuationsAsynchronously).
